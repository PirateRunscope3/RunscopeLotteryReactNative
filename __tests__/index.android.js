import 'react-native';
import React from 'react';
import Index from '../index.android.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Index />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('getNumber', () => {
  const lotteryNumbers = new Index();
  const num = lotteryNumbers.getNumber();
  expect(num).toBeGreaterThan(0);
  expect(num).toBeLessThan(50);
});

it('generateNumbers', () => {
  const lotteryNumbers = new Index();
  const nums = lotteryNumbers.generateNumbers().split('  ');
  var set = new Set();

  expect(nums.length).toBe(6);

  for (var i = 0; i < 6; i++) {
    expect(Number(nums[i])).toBeGreaterThan(0);
    expect(Number(nums[i])).toBeLessThan(50);
    expect(set.has(nums[i])).toBeFalsy();
    set.add(nums[i]);
  }
});

