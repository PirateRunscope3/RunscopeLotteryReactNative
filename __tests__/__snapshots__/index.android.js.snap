exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }>
        Generate Numbers!
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "fontSize": 20,
        "textAlign": "center",
      }
    }>
    
    
    
  </Text>
</View>
`;
